src/backup.ts

```diff
@@ -0,0 +1,373 @@
+/**
+ * Copyright 2020 Google LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {promisifyAll} from '@google-cloud/promisify';
+import {google as databaseAdmin} from '../proto/spanner_database_admin';
+import {Instance} from './instance';
+import {RequestCallback, ResourceCallback} from './common';
+import {EnumKey, RequestConfig, TranslateEnumKeys} from '.';
+import {Metadata, Operation as GaxOperation} from 'google-gax';
+import * as extend from 'extend';
+import {DateStruct, PreciseDate} from '@google-cloud/precise-date';
+import {status} from 'grpc';
+
+export type CreateBackupCallback = ResourceCallback<
+  GaxOperation,
+  databaseAdmin.longrunning.IOperation
+>;
+
+export interface CreateBackupGaxOperation extends GaxOperation {
+  // Overridden with more specific type for CreateBackup operation
+  metadata: Metadata &
+    databaseAdmin.spanner.admin.database.v1.ICreateBackupMetadata;
+}
+
+export type CreateBackupResponse = [
+  CreateBackupGaxOperation,
+  databaseAdmin.longrunning.IOperation
+];
+
+/**
+ * IBackup structure with backup state enum translated to string form.
+ */
+type IBackupTranslatedEnum = TranslateEnumKeys<
+  databaseAdmin.spanner.admin.database.v1.IBackup,
+  'state',
+  typeof databaseAdmin.spanner.admin.database.v1.Backup.State
+>;
+
+export type GetMetadataResponse = [IBackupTranslatedEnum];
+type GetMetadataCallback = RequestCallback<IBackupTranslatedEnum>;
+
+type UpdateExpireTimeCallback = RequestCallback<Backup>;
+
+type DeleteCallback = RequestCallback<void>;
+
+/**
+ * The {@link Backup} class represents a Cloud Spanner backup.
+ *
+ * Create a `Backup` object to interact with or create a Cloud Spanner backup.
+ *
+ * @class
+ *
+ * @example
+ * const {Spanner} = require('@google-cloud/spanner');
+ * const spanner = new Spanner();
+ * const instance = spanner.instance('my-instance');
+ * const backup = instance.backup('my-backup');
+ */
+class Backup {
+  request: <T, R = void>(
```

Comment:
There could be three types of callbacks for request
 - ResourceCallback, in case when longrunning Operation is returned as in createBackup
    - The request callback will return (err, operation: [GaxOperation](https://github.com/googleapis/nodejs-spanner/blob/2a7144b2e31f077df6085ae256e5a41a680d69e3/src/database.ts#L31), apiResponse: IOperation)
 - NormalCallback as in getBackup
   - Where the request callback will return (err, resp: google.spanner.admin.database.v1.IBackup)
 - PagedCallback as in listBackups
   - Where the request callback will return (err, results: protoBackupObjects, nextQuery (next page request), reponse: IListBackupsResponse )

Reply:
I think we're following the setup in [`database.ts`](https://github.com/googleapis/nodejs-spanner/blob/2a7144b2e31f077df6085ae256e5a41a680d69e3/src/database.ts#L203). The [`instance.ts`](https://github.com/googleapis/nodejs-spanner/blob/2a7144b2e31f077df6085ae256e5a41a680d69e3/src/instance.ts#L94) setup looks slightly different as well.

Can you show a code example of how this is meant to be setup?

Reply:
I finally understood this and was able to fix the issues. Now all the classes have the callback types setup correctly. The only one I didn't configure was the PagedCallback for Instance since all the existing get*s() functions in that class use a NormalCallback.

Reply:
Now using PagedCallback for getBackups(), getBackupOperations() and getDatabaseOperations().